src/libraries/filcrypto/filproofs/algorithms.id

```diff
@@ -112,14 +112,23 @@ type WinStackedDRG struct {
     ) bool
     VerifySeal(sv sector.SealVerifyInfo) bool
 
-    GeneratePoStCandidates(
+    GenerateElectionPoStCandidates(
         challengeSeed    sector.PoStRandomness
         eligibleSectors  [sector.SectorNumber]
         candidateCount   int
         sectorStore      sectorIndex.SectorStore
-    ) [sector.ChallengeTicket]
-    GeneratePoStProof(privateProofs [sector.PrivatePoStProof]) sector.PoStProof
-    VerifyPoSt(sv sector.PoStVerifyInfo) bool
+    ) [sector.PoStCandidate]
+    GenerateElectionPoStProof(privateProofs [sector.PrivatePoStProof]) sector.PoStProof
+    VerifyElectionPoSt(sv sector.PoStVerifyInfo) bool
+
+    GenerateSurprisePoStCandidates(
```

Comment:
why do we have SurprisePoStCandidates and SurprisePoStProofs? I thought it is one single method but I can be totally wrong on this.

Reply:
no it's a two time operation since you will only generate a proof on a portion of your candidates (the winning ones) in ePost